hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -278,6 +280,21 @@ protected final StoreFileWriter createTmpWriter(FileDetails fd, boolean shouldDr
         HConstants.EMPTY_STRING);
   }
 
+  /**
+   * Default method for initializing a StoreFileWriter in the compaction process, this creates the
```

Comment:
Is this right if we have the StoreEngine in the mix? Its supposed to do this right?

Reply:
StoreEngine delegates the work to a `Compactor` implementation, where `DefaultCompator`, the default Compactor implementation, uses this one to create the file writer. So that's why we say it's the "default" method. Alternatives compactors may not use it, like the `DirectStoreCompactor` that implements its own logic that creates the file on the store directory, rather than under temp.

Reply:
>Is creating them in the temp directory really "default"? Or is it "default" just because the direct-insert stuff isn't available yet? ;)

It is the default now, if we don't explicitly specify `hbase.hstore.defaultengine.compactor.class` config property to set `DirectStoreCompactor` implementation to be used, the previous existing `DefaultCompactor` will be loaded, which implements this logic.

> When you say other compactors may not use it, it makes me wonder if it makes sense to provide it here.

What I meant is that other compactors may overwrite the method with a different logic.

Reply:
> What I meant is that other compactors may overwrite the method with a different logic.

I'm just thinking about what other potential compaction algorithms may choose to do. I would expect that any future compactions would still want to leverage the "commit logic" here to avoid that rename. Just keeping tabs on "pre-optimization" without a specific use-case in mind. Although, as long as other compaction algorithms can extend DirectStoreCompactor and get the new shiny solution, it's not a concern.